core/store/src/trie/trie_storage.rs

```diff
@@ -72,7 +71,7 @@ pub struct TrieCacheInner {
     /// Upper bound for the total size.
     total_size_limit: u64,
     /// Shard id of the nodes being cached.
-    shard_id: u32,
+    shard_id: ShardId,
```

Comment:
:+1: :+1: :+1:

Reply:
Yeah this is one is weirder than it might look. `ShardId` is `u64` but in NEP241 we specified `shard_id` inside `ShardUId` to be `u32`, such that it fits in 8 bytes. Changing everything to `u32` doesn't work, as there are places where we deserialize it from `u64`.
I considered creating an issue on GH for a general cleanup. But I'm not sure there is anything worth while doing at this point. Probably it's best to just use `ShardId = u64` for `shard_id` everywhere except inside `ShardUId`.